pkgs/by-name/ne/newlisp/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  lib,
+  fetchurl,
+  stdenv,
+  libffi,
+  readline,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "newlisp";
+  version = "10.7.5";
+
+  src = fetchurl {
+    url = "https://www.newlisp.org/downloads/newlisp-${finalAttrs.version}.tgz";
```

Comment:
I love the use of finalAttrs.version but note that it will always fail the embedded hash here, unless you allow overriding that, too. Which is probably overkill since it's easier for anyone who wants a different version to just bring their own src and inject that, but it's worth noting that this doesn't actually have the effect it suggests it will.

Reply:
> unless you allow overriding that

It is possible to override hash separately, see https://github.com/NixOS/nixpkgs/issues/310373#issuecomment-2209336570

Reply:
I do this just for reusing the `version` while keeping the scope clean (unlike the rough `rec`). Never thought about someone overriding `version` only. 😶

Reply:
Nice yeah good point you can do a nested override. Fair enough!